### PR TITLE
Joblib update to default -1

### PR DESCRIPTION
**Pull request by Ziqi-Li**
Update to `n_jobs = -1` as default to use all processors. And to fix typo and make clarification in the example notebook.

### CODE REVIEW

**Comment by martinfleis** [comment]
Thanks!

What about that `pool` keyword causing hard-breaking change (https://github.com/pysal/mgwr/pull/137#discussion_r1414313848)? That was the only point I though actually needs to be resolved, but these are nice to have as well!

**Comment by Ziqi-Li** [comment]
> Thanks!
> 
> What about that `pool` keyword causing hard-breaking change ([#137 (comment)](https://github.com/pysal/mgwr/pull/137#discussion_r1414313848))? That was the only point I though actually needs to be resolved, but these are nice to have as well!

How about this one with the new commit?
image

File: mgwr/gwr.py [hunk 1]
```diff
@@ -333,6 +335,9 @@ def fit(self, ini_params=None, tol=1.0e-5, max_iter=20, solve='iwls',
         self.fit_params['solve'] = solve
         self.fit_params['lite'] = lite
 
+        if pool:
+            warnings.warn("The pool parameter is no longer used and will have no effect; parallelization is default and implemented using joblib instead.", RuntimeWarning)
```
Comment (martinfleis) [inline mgwr/gwr.py:339]:
```suggestion
            warnings.warn("The pool parameter is no longer used and will have no effect; parallelization is default and implemented using joblib instead.", RuntimeWarning, stacklevel=2)
```

File: mgwr/gwr.py [hunk 2]
```diff
@@ -1631,6 +1638,9 @@ def tqdm(x, total=0,
                      desc=''):  #otherwise, just passthrough the range
                 return x
 
+        if pool:
+            warnings.warn("The pool parameter is no longer used and will have no effect; parallelization is default and implemented using joblib instead.", RuntimeWarning)
```
Comment (martinfleis) [inline mgwr/gwr.py:1642]:
```suggestion
            warnings.warn("The pool parameter is no longer used and will have no effect; parallelization is default and implemented using joblib instead.", RuntimeWarning, stacklevel=2)
```

File: mgwr/sel_bw.py [hunk 3]
```diff
@@ -290,6 +292,10 @@ def search(self, search_method='golden_section', criterion='AICc',
                 " a single entry or a list containing an entry for each of k"
                 " covariates including the intercept")
 
+        if pool:
+            warnings.warn("The pool parameter is no longer used and will have no effect; parallelization is default and implemented using joblib instead.", RuntimeWarning)
```
Comment (martinfleis) [inline mgwr/sel_bw.py:296]:
```suggestion
            warnings.warn("The pool parameter is no longer used and will have no effect; parallelization is default and implemented using joblib instead.", RuntimeWarning, stacklevel=2)
```

**Review by martinfleis — APPROVED** [review]
Minor nits. The rest looks great. Thanks!

**Pull request merged**